src/read.c

```diff
@@ -1521,7 +1500,7 @@ static avifBool avifParseItemLocationBox(avifMeta * meta, const uint8_t * raw, s
     }
     for (uint32_t i = 0; i < itemCount; ++i) {
         uint32_t itemID;
-        uint32_t idatID = 0;
+        avifBool idatStored = AVIF_FALSE;
```

Comment:
This local variable does not seem necessary. Why don't we just set `item->idatStored` directly?

Reply:
We have not yet called `avifMetaFindItem()`, which is what lazily creates the item structure. We could create it before the `constructionMethod` block to avoid the temporary variable, I suppose. I believe my original intent was to avoid calling `avifMetaFindItem()` until I was sure the data is as valid as possible, but we have plenty of potential errors after this call, so it isn't a big deal to move it earlier.

Reply:
I've added a second commit to the PR (so you can see the specific change for this). Do you prefer this?

Reply:
I didn't know `item` was created later. This is fine then.

Reply:
No need to move the `avifMetaFindItem()` call unless you prefer that.